pkg/cloud/openstack/openstack.go

```diff
@@ -125,12 +125,14 @@ func CloudConfigTransformer(source string, infra *configv1.Infrastructure) (stri
 		}
 	}
 
-	for key, value := range map[string]string{
+	configKeys := []string{"cloud", "clouds-file", "use-clouds"}
+	configMap := map[string]string{
```

Comment:
@pierreprinetti is going to hate this too, but how about:

```golang
for _, o := range []struct{ k, v string } {
  {"use-clouds", "true"},
  ...
} {
  _, err = global.NewKey(o.k, o.v)
  ...
}
```

Reply:
Yeah, we'll need to use a slice to preserve the keys order.
We should perhaps add a comment too to make it more obvious:

```go
// Use a slice to preserve keys order
for _, o := range []struct{ k, v string }{
	{"use-clouds", "true"},
	{"clouds-file", "/etc/openstack/secret/clouds.yaml"},
	{"cloud", "openstack"},
} {
	_, err = global.NewKey(o.k, o.v)
	if err != nil {
		return "", fmt.Errorf("failed to modify the provided configuration: %w", err)
	}
}

```